src/range/Range1D.ts

```diff
@@ -204,28 +204,59 @@ export default class Range1D {
   }
 
   /**
-   * logical union between two ranges
+   * merge list of two ranges
    * @param other
-   * @returns {RangeDim}
+   * @param size
+   * @returns {number[]}
+   * @private
    */
-  union(other: Range1D, size?: number): Range1D {
-    if (this.isAll || other.isNone) {
-      return this.clone();
-    }
-    if (other.isAll || this.isNone) {
-      return other.clone();
-    }
+  private _union(other: Range1D, size?: number): number[] {
     const r = this.iter(size).asList();
     const it2 = other.iter(size);
     it2.forEach((i) => {
       if (r.indexOf(i) < 0) {
         r.push(i);
       }
     });
-    return other.fromLike(r.sort());
+    return r;
+  }
+
+  /**
+   * sorted logical union between two ranges
+   * @param other
+   * @param size
+   * @returns {Range1D}
+   */
+  union(other: Range1D, size?: number): Range1D {
+    if (this.isAll || other.isNone) {
+      return this.clone();
+    }
+    if (other.isAll || this.isNone) {
+      return other.clone();
+    }
+    const r = this._union(other, size).sort(); // sorted after making union
+    return other.fromLike(r);
   }
 
   /**
+   * concatenate this range (first) with another range (second) without sorting
+   * @param other
+   * @param size
+   * @returns {Range1D}
+   */
+  concat(other: Range1D, size?: number): Range1D {
```

Comment:
there is already a join method: https://github.com/phovea/phovea_core/blob/develop/src/range/index.ts#L50